lib/inputs/distroless/static.ts

```diff
@@ -0,0 +1,21 @@
+import { ExtractAction, ExtractedLayers } from "../../extractor/types";
+import { streamToString } from "../../stream-utils";
+
+export const getDpkgPackageFileContentAction: ExtractAction = {
+  actionName: "dpkg",
+  fileNamePattern: "/var/lib/dpkg/status.d/*",
+  callback: streamToString, // TODO replace with a parser for apt data extractor
```

Comment:
this will make the difference between loading into our memory all the files in ```/var/lib/dpkg/status.d/*``` (in addition to all the files we normally load) and only afterwards extracting the little that we need, or loading one file at a time and storing what we need as we go along.
should be significant proportionally to the number of installed packages.

Reply:
Interesting! So the callback would be a function that processes the stream in place and returns only the data that we need from the file? (Instead of reading the content as a string, and only later going over the string to figure out what we need)

Reply:
thinking of a function that first collects the entire file as a string, then processes it, and stores only the interesting bits.
that is, instead of resulting with a list of files with their entire content, just do the analysis on each file after reading it.
haven't thought it 100% through but the idea is just not storing all the files in memory all the time.